### ISSUE TITLE
Minor accuracy bug

### ISSUE THREAD

**Issue by stfwn** [stfwn/ats-privacy-replication#43]
## Summary

Accuracy is computed per batch and averaged over the batch dimension, resulting in about 0.2% error in practice versus computing it per sample.

## Details

Accuracy is computed per batch here:

https://github.com/stfwn/mscai-fact-ai/blob/d838a4baf957fcb59e6ed702566ccaf8a9af974f/ATSPrivacy/inversefed/data/loss.py#L120-L128

All the batch accuracies are summed up:

https://github.com/stfwn/mscai-fact-ai/blob/d838a4baf957fcb59e6ed702566ccaf8a9af974f/ATSPrivacy/inversefed/training/training_routine.py#L88-L89

And averaged to produce the epoch metric:

https://github.com/stfwn/mscai-fact-ai/blob/d838a4baf957fcb59e6ed702566ccaf8a9af974f/ATSPrivacy/inversefed/training/training_routine.py#L101

But since the test set is 10k samples big, batch size is 128 and `drop_last=False` in the dataloader, this leads to one last batch of `(10000 / 128 - 10000 // 128) * 128 = 16` samples counting `128 / 6 = 8` times more than other samples.

In the worst case those 16 are correct and all hypothetical other samples of the last batch would be wrong, leading to one 100% in the tally versus a 12.5%. If all the other samples have 70% accuracy this results in a score of a 1% percent higher accuracy. (`((78*0.7+1)/79 - (78*0.7+0.125)/79)*100`)

In practice it only matters about 0.2% on average.
